Lib/test/test_sysconfig.py

```diff
@@ -620,10 +620,17 @@ def test_abiflags(self):
         self.assertIsInstance(ABIFLAGS, str)
         self.assertIn(abiflags, ABIFLAGS)
 
+        # Check all flags are alphabetic
         if os.name == 'nt':
             self.assertEqual(abiflags, '')
             # Example values: '', 't', 't_d', '_d'
-            self.assertTrue(ABIFLAGS.count('_d') == 1 or '_' not in ABIFLAGS, ABIFLAGS)
+            # '_' can only exist if 'd' is present
+            if '_' in ABIFLAGS:
+                # '_' is followed by 'd'
+                self.assertTrue(ABIFLAGS.count('_d') == 1, ABIFLAGS)
+                self.assertTrue(ABIFLAGS.count('_') == 1, ABIFLAGS)
+            else:
+                self.assertNotIn('d', ABIFLAGS)
```

Comment:
```suggestion
            self.assertIn(ABIFLAGS, ('', '_d', 't', 't_d'))
```

I was thinking of this. Your test is just as constrained to these cases, so may as well be very direct - there's already a note about updating the test if the behaviour changes.

Reply:
> Your test is just as constrained to these cases, so may as well be very direct

I'm thinking of potential future changes to the ABI flags, such as adding a new flag or removing the `t` flag. Expilictly enumerating the current flag candidates will need extra maintenance effort in the future.

Reply:
It’s good that test_sysconfig needs to be changed if ABI flags change in the future.
I agree with Steve here.